geochem_inverse_optimize.py

```diff
@@ -255,7 +304,10 @@ def solve_montecarlo(
             )  # Solve problem
             for sample_name in element_pred_down.keys():
                 predictions_down_mc[sample_name] += [element_pred_down[sample_name]]
-                predictions_up_mc[sample_name] += [element_pred_upstream[sample_name]]
+                if not self.continuous:
```

Comment:
Might be better to factor this `if` out of the for-loop so that the for loop does a single thing and we have an `if-else` on continuous.